cpp/cmake_modules/Findutf8proc.cmake

```
@@ -15,10 +15,16 @@
 # specific language governing permissions and limitations
 # under the License.
 
+if(ARROW_UTF8PROC_USE_SHARED)
+  set(UTF8PROC_NAMES utf8proc)
+else()
+  set(UTF8PROC_NAMES utf8proc utf8proc_static)
+endif()
```

How about more strict logic like our `FindZSTD.cmake`?
https://github.com/apache/arrow/blob/master/cpp/cmake_modules/Fondest.cmake#L18-L34

Because `_static` suffix is only appended for MSVC: https://github.com/JuliaStrings/utf8proc/blob/master/CMakeLists.txt#L32-L34

```suggestion
if(ARROW_UTFPROC8_USE_SHARED)
  set(UTFPROC8_NAMES)
  if(CMAKE_IMPORTRARY_SUFFIX)
    list(APPEND UTFPROC8_NAMES
                "${CMAKE_IMPORT_LIBRARY_PREFIX}utfproc8${CMAKE_IMPORT_LIBRARY_SUFFIX}")
  endif()
  list(APPEND UTFPROC8_NAMES
              "${CMAKE_SHARED_LIBRARY_PREFIX}utfproc8${CMAKE_SHARED_LIBRARY_SUFFIX}")
else()
  if(MSVC AND NOT DEFINED UTFPROC8_MSVC_STATIC_LIB_SUFFIX)
    set(UTFPROC8_MSVC_STATIC_LIB_SUFFIX "_static")
  endif()
  set(UTFPROC8_STATIC_LIB_SUFFIX
      "${UTFPROC8_MSVC_STATIC_LIB_SUFFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}")
  set(UTFPROC8_NAMES "${CMAKE_STATIC_LIBRARY_PREFIX}utfproc8${UTFPROC8_STATIC_LIB_SUFFIX}")
endif()
```